R/estimateDecoyFDR.R

```diff
@@ -1,95 +1,152 @@
+  # setting a seed is absolutely crutial to ensure reproducible results!
   clusterSetRNGStream(cl,123)
   doSNOW::registerDoSNOW(cl)
   clusterEvalQ(cl,library(SECprofiler,data.table))
-  data <- list()
-  data <- c(data,parRapply(cl,parameter_grid,FUN=runGridComplexFeatureFinding,protTraces=traces,calibration=calibration,complex_hypothesis=complex_hypothesis))
+  data <- parRapply(cl,parameter_grid,
+                           FUN = .runGridComplexFeatureFinding,
+                           protTraces = traces,
+                           complex_hypothesis = complex_hypothesis)
   stopCluster(cl)
-  data
+
+  return(data)
 }
 
-runGridComplexFeatureFinding <- function(params,protTraces,calibration,complex_hypothesis) {
-  res = findComplexFeatures(traces=protTraces,
-                            complex_hypothesis = complex_hypothesis,
-                            calibration = calibration,
-                            corr_cutoff = as.numeric(params["corr"]),
-                            window_size = as.numeric(params["window"]),
-                            parallelized = FALSE,
-                            collapse_method="apex_network",
-                            perturb_cutoff="1%",
-                            rt_height=as.numeric(params["rt_height"]),
-                            smoothing_length=as.numeric(params["smoothing"])
-                            )
-  res[,corr:=as.numeric(params["corr"])]
-  res[,window:=as.numeric(params["window"])]
-  res[,rt_height:=as.numeric(params["rt_height"])]
-  res[,smoothing_length:=as.numeric(params["smoothing"])]
+.runGridComplexFeatureFinding <- function(params,
+                                         protTraces,
+                                         complex_hypothesis) {
+  res <- findComplexFeatures(traces=protTraces,
+                             complex_hypothesis = complex_hypothesis,
+                             corr_cutoff = as.numeric(params["corr"]),
+                             window_size = as.numeric(params["window"]),
+                             parallelized = FALSE,
+                             collapse_method="apex_network",
+                             perturb_cutoff="1%",
+                             rt_height=as.numeric(params["rt_height"]),
+                             smoothing_length=as.numeric(params["smoothing"]))
+  
+  res$corr <- as.numeric(params["corr"])
+  res$window <- as.numeric(params["window"])
+  res$rt_height <- as.numeric(params["rt_height"])
+  res$smoothing_length <- as.numeric(params["smoothing"])
   #saveRDS(res,file=paste0("complexFeatures/complexFeatures_",name,".rda"))
   res[]
 }
 
-#' Perform complex feature grid search filter
-#' @description Perform complex feature grid search filter.
-#' @param complex_features data.table containing filtered complex feature results.
-#' @return List with stats
+#' Perform complex feature grid search filtering
```

Comment:
Please add proper description and examples to all of the following functions that are exported!